frontend/app/routes/$lang/_public/status/child/index.tsx

```diff
@@ -195,15 +195,20 @@ export async function action({ context: { session }, params, request }: ActionFu
   const applicationStatusService = getApplicationStatusService();
   const lookupService = getLookupService();
 
-  // TODO: Change getStatusId call for the new API call. (Coming soon) **Update: Email has been received. Working on it.
-  // TODO: Create 2 functions - getChildStatusIdBySin & getChildStatusIdByPersonalInfo
-  const statusId = await applicationStatusService.getStatusId({
-    sin: parsedSinResult?.data.sin ?? '',
-    applicationCode: parsedCodeResult.data.code,
-    firstName: parsedChildInfoResult?.data.firstName ?? '',
-    lastName: parsedChildInfoResult?.data.lastName ?? '',
-    dateOfBirth: parsedChildInfoResult?.data.dateOfBirth ?? '',
-  });
+  let statusId: string | null;
+  if (!parsedSinResult) {
+    statusId = await applicationStatusService.getStatusIdWithoutSin({
+      applicationCode: parsedCodeResult.data.code,
+      firstName: parsedChildInfoResult?.data.firstName ?? '',
+      lastName: parsedChildInfoResult?.data.lastName ?? '',
+      dateOfBirth: parsedChildInfoResult?.data.dateOfBirth ?? '',
+    });
+  } else {
+    statusId = await applicationStatusService.getStatusIdWithSin({
+      sin: parsedSinResult.data.sin,
+      applicationCode: parsedCodeResult.data.code,
+    });
+  }
```

Comment:
**Nitpick:** Prefer the use of `const` to make it clear that `statusId` won't be reassigned:

```typescript
const statusId = parsedSinResult
  ? await applicationStatusService.getStatusIdWithSin({
      sin: parsedSinResult.data.sin,
      applicationCode: parsedCodeResult.data.code,
    })
  : await applicationStatusService.getStatusIdWithoutSin({
      applicationCode: parsedCodeResult.data.code,
      firstName: parsedChildInfoResult?.data.firstName ?? '',
      lastName: parsedChildInfoResult?.data.lastName ?? '',
      dateOfBirth: parsedChildInfoResult?.data.dateOfBirth ?? '',
    });
```